ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -85,10 +85,13 @@ def run_checks() -> Tuple[Set[str], int, int]:
             result = check_instance.check()
         except Exception:
             doctor_warn('Fail to call %s class functions.' % check_entry_pt.name)
-        if result is False:
-            fail += 1
-            failed_cats.add(check_category)
-        total += 1
+        if isinstance(result, bool):
```

Comment:
Doesn't this lead to a `UnboundLocalError` if `check_instance.check()` throws an exception?  Actually, worse than that, if *this* loop iteration throws an exception, it will actually re-check the value of the *last* iteration `result`.  The simple way to fix this may be to set `result` to `False` at the top of the loop, and then the rest of this code will work.